### DISCUSSION TITLE
What is the tool for working with admin.sqlite data?

### DISCUSSION THREAD

**Discussion by nvcnvn** [valhalla/valhalla#4311] [Q&A] [answered]
I'm looking at the test data https://github.com/valhalla/valhalla/blob/master/test/data/netherlands_admin.sqlite here
What is the tool/standard for working with these kind of data?

Thanks :)

**Comment by ianthetechie** [comment, marked as answer]
What sort of work do you want to do with it?

Here's a bit of background on what this is/does... It's a database of administrative regions and some basic info about them. Why do we need it? A few reasons, but the most obvious that come to mind are determining whether we drive on the right or left side of the road, and identifying country border crossings.

How is it generated? The `valhalla_build_admins` utility creates this file from an OpenStreetMap PBF file.

Where is it used? Only when building the routing tiles. The necessary information is added to the routing graph during the tile build. This file is not used by a running server.

How can you "work with" this data? It depends on what you want to do I guess. It's a SQLite database, specifically one with the Spatialite extension. You can use any tool for manipulating SQLite files like the `sqlite3` CLI utility, DataGrip, or probably any other commercial database explorer. You can also probably load it up in QGIS if you wanted a more geospatial-oriented workflow and visualize the data on a basemap and do GIS-y things ™️ 

Hope this helps!

**Reply by nvcnvn** [reply]
I'm working with the speed assignment based on iso3166-1 and iso3166-2 config (https://github.com/valhalla/valhalla/discussions/4307)
I was thinking to the admin data is something I need to import to make that function work.
Not sure if my above assumption is correct? (from your comment the boundary should be included in the pbf already?)

**Reply by nilsnolde** [reply]
Yes, you'll need to run `valhalla_build_admins` on the same PBF as you'll build the graph with (but before building the graph of course), as @ianthetechie suggests. So you had the right hunch.
